1. Basic Web Development/frontend/src/main.ts

```diff
@@ -409,11 +502,15 @@ function toggleLoad(isLoading: boolean) {
 }
 
 class Permission {
-    id: string;
+    _id: string;
 
     constructor(public text: string, public type: string) {
```

Comment:
As previously discussed, type should be an enum or you should fetch available permission types from be.